### PR TITLE
#431 further changes to maintain backward compatiblility

### PR DESCRIPTION
**Pull request by Emily-Jiang**
Signed-off-by: Emily Jiang <emijiang6@googlemail.com>

### CODE REVIEW

File: api/src/main/java/org/eclipse/microprofile/config/spi/ConfigSource.java [hunk 1]
```diff
@@ -83,63 +84,76 @@
...
 
     /**
-     * Gets all property names known to this config source, potentially without evaluating the values.
+     * Gets all property names known to this config source, potentially without
+     * evaluating the values.
      *
-     * For backwards compatibility, there is a default implementation that just returns the keys of {@code getProperties()}
-     * slower ConfigSource implementations should replace this with a more performant implementation
+     * For backwards compatibility, there is a default implementation that just returns
+     * an empty set. This method should be overriden.
```
Comment (ljnelson) [inline api/src/main/java/org/eclipse/microprofile/config/spi/ConfigSource.java]: Minor, but I would prefer this documentation to read something like the following:

    The default implementation of this method returns an empty {@link Set}.
    {@link ConfigSource} implementations should override this method.

File: api/src/main/java/org/eclipse/microprofile/config/spi/ConfigSource.java [hunk 2]
```diff
@@ -83,63 +84,76 @@
...
     }
 
     /**
-     * Gets all property names known to this config source, potentially without evaluating the values.
+     * Gets all property names known to this config source, potentially without
+     * evaluating the values.
      *
-     * For backwards compatibility, there is a default implementation that just returns the keys of {@code getProperties()}
-     * slower ConfigSource implementations should replace this with a more performant implementation
+     * For backwards compatibility, there is a default implementation that just returns
```
Comment (ljnelson) [inline api/src/main/java/org/eclipse/microprofile/config/spi/ConfigSource.java]: Javadoc improvements: normally every paragraph after the first sentence should be wrapped in `<p>` tags.

File: api/src/main/java/org/eclipse/microprofile/config/spi/ConfigSource.java (hunk 1 above)
Comment (Emily-Jiang) [reply api/src/main/java/org/eclipse/microprofile/config/spi/ConfigSource.java]: done

**Comment by tevans78** [comment]
This change is not backward compatible with exsiting older ConfigSource implementations which were relying on the default implementation of getPropertyNames. A call to Config.getPropertyNames() would previously returned a complete set based on the keys from ConfigSource.getProperties(). Now, if ConfigSource.getPropertyNames() is not overridden, that set will not be complete.

**Comment by Emily-Jiang** [comment]
Not going to pursue this one any further, so I closed it.

**Pull request closed without merging**